### PR TITLE
ATB-1243: Enable API Gateway logging along with CSLS subscription

### PR DESCRIPTION
**Pull request by liamquayle**
## Proposed changes
[ATB-1243](https://govukverify.atlassian.net/browse/ATB-1243)

### What changed
Ensuring API GW Rest API log group has a subscription to CSLS

### Why did it change
Need to send logs to Cyber monitoring tool (Splunk)

## Testing
Currently no access to splunk to check if logs are successfully sent, will update this in due course

AccessLogGroup was successfully created, as seen in API GW:
Screenshot 2023-12-13 at 10 59 17

Created log group for Invoke Private API function:
Screenshot 2023-12-13 at 16 10 56

Successfully publishes logs to the log group when function is invoked:
Screenshot 2023-12-13 at 16 14 51

## Checklists
- [x] Did not commit any not-required changes to the `src/infra/**/samconfig.toml`
- [ ] Tested changes and included test evidence in the PR, if appropriate
- [x] Included all required tags and other properties for any new resources in the SAM template
- [x] Ensured that any new resources in the SAM Template follow appropriate naming conventions
- [x] Ensured that naming of new resources is compatible with deploying multiple stacks with custom stack names during development
- [x] Ensured that no log lines include PII or other sensitive data
- [ ] Implemented unit testing for any new logic implemented, if appropriate
- [x] Ensured that all commits in this PR are signed
- [ ] Ensured appropriate code coverage is maintained by unit tests

[ATB-1243]: https://govukverify.atlassian.net/browse/ATB-1243?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/infra/main/template.yaml [hunk 1]
```diff
@@ -900,32 +964,16 @@ Resources:
...
-          - Effect: Allow
-            Action:
-              - "kms:decrypt"
-            Resource:
-              - Fn::ImportValue: !Sub "${CoreStackName}-AccountStatusTableEncryptionKeyArn"
-      Roles:
-        - !Ref StatusRetrieverFunctionRole
+      DestinationArn: !Ref CSLSSubscriptionEndpointArn
+      LogGroupName: "/aws/lambda/InvokePrivateAPIGatewayFunction"
+      FilterPattern: "{$.message != \"Sensitive info*\"}"
```
Comment (chiarapascucci-gds) [inline src/infra/main/template.yaml:976]: I don't think we need this filter pattern here, as we are not adding this prefix in api gateway logs

File: src/infra/main/template.yaml [hunk 2]
```diff
@@ -996,27 +1044,49 @@ Resources:
...
+        Owner: !Ref OwnerTagValue
+        Source: !Ref SourceTagValue
+
+  PrivateAPIGatewayLogsSubscriptionFilterCSLS:
+    Condition: IsProdOrInt
+    Type: AWS::Logs::SubscriptionFilter
+    Properties:
+      DestinationArn: !Ref CSLSSubscriptionEndpointArn
+      LogGroupName: !Ref PrivateAPIGatewayAccessLogGroup
+      FilterPattern: ""
```
Comment (chiarapascucci-gds) [inline src/infra/main/template.yaml:1089]: is it possible to define the resource without this property? I am not sure how an empty string would be interpreted here

File: src/infra/main/template.yaml [hunk 3]
```diff
@@ -996,27 +1044,49 @@ Resources:
...
+        - Key: Name
+          Value: PrivateApiGatewayAccessLogGroup
+        - Key: Owner
+          Value: !Ref OwnerTagValue
+        - Key: Source
+          Value: !Ref SourceTagValue
+        - Key: Name
+          Value: !Sub "/aws/apigateway/${PrivateRestApi}/${ApiStageName}"
+
+  PrivateAPIAccessLogGroupNameSSM:
```
Comment (chiarapascucci-gds) [inline src/infra/main/template.yaml:1069]: Do we need this resource?

File: src/infra/main/template.yaml (hunk 2 above)
Comment (liamquayle) [reply src/infra/main/template.yaml:1089]: Happy enough to remove this if needed, copied line 2837 where the resource for the CSLS subscription uses an empty string as a filter pattern, for the vc storage private api access logs.

File: src/infra/main/template.yaml (hunk 3 above)
Comment (liamquayle) [reply src/infra/main/template.yaml:1069]: dont think so, passes linting and deploys without so took it out

File: src/infra/main/template.yaml (hunk 2 above)
Comment (liamquayle) [reply src/infra/main/template.yaml:1089]: turns out `FilterPattern` is required from the linter.

**Review by chiarapascucci-gds — APPROVED** [review]

File: src/infra/main/template.yaml [hunk 4]
```diff
@@ -996,27 +1044,35 @@ Resources:
...
+          Value: PrivateApiGatewayAccessLogGroup
+        - Key: Owner
+          Value: !Ref OwnerTagValue
+        - Key: Source
+          Value: !Ref SourceTagValue
+        - Key: Name
+          Value: !Sub "/aws/apigateway/${PrivateRestApi}/${ApiStageName}"
+
+  PrivateAPIGatewayLogsSubscriptionFilterCSLS:
+    Condition: IsProdOrInt
```
Comment (TaherKapasi) [inline src/infra/main/template.yaml:1070]: remove condition

**Review by TaherKapasi — CHANGES_REQUESTED** [review]

**Review by TaherKapasi — APPROVED** [review]

**Review by TaherKapasi — CHANGES_REQUESTED** [review]
Hi @liamquayle - could we add the Condition: IsInternal to the InvokeLambda role as well please

**Review by TaherKapasi — APPROVED** [review]

**Pull request merged**